lib/mfg.c

```diff
@@ -411,6 +411,99 @@ int switchtec_secure_state_set(struct switchtec_dev *dev,
 			     NULL, 0);
 }
 
+/**
+ * @brief Unlock firmware debug features
+ * @param[in]  dev		Switchtec device handle
+ * @param[in]  serial		Device serial number
+ * @param[in]  ver_sec_unlock	Secure unlock version
+ * @param[in]  public_key	public key data
+ * @param[in]  signature	Signature of data sent
+ * @return 0 on success, error code on failure
+ */
+int switchtec_dbg_unlock(struct switchtec_dev *dev, uint32_t serial,
+			 uint32_t ver_sec_unlock,
+			 struct switchtec_pubkey *public_key,
+			 struct switchtec_signature *signature)
+{
+	int ret;
+	struct unlock_cmd1 {
+		uint32_t subcmd;
+		uint8_t pub_key[SWITCHTEC_PUB_KEY_LEN];
+		uint32_t pub_key_exp;
+	} cmd1 = {};
+
+	struct unlock_cmd2 {
+		uint32_t subcmd;
+		uint32_t serial;
+		uint32_t unlock_ver;
+		uint8_t signature[SWITCHTEC_SIG_LEN];
+	} cmd2 = {};
+
+	cmd1.subcmd = htole32(MRPC_DBG_UNLOCK_PKEY);
```

Comment:
Seems like MRPC_DBG_UNLOCK_PKEY should be it's own static function so it can be reused in both of these functions.